pkg/parser/custom/py/setuppy_parser.go

```diff
@@ -0,0 +1,233 @@
+package py 
+
+import (
+	"fmt"
+	"os"
+	"strings"
+	"context"
+	// "github.com/mpvl/unique"
+	"github.com/safedep/vet/pkg/common/logger"
+	treesitter "github.com/smacker/go-tree-sitter"
+	"github.com/smacker/go-tree-sitter/python"
+)
+
+// SetuppyParserViaSyntaxTree holds the parsing and extraction logic for setup.py files.
+type SetuppyParserViaSyntaxTree struct
```

Comment:
Do we need to expose this outside the `py` package because the name indicate it is specific to `py` package